### PR TITLE
Jetpack Cloud: Disable downloadable backup creation while another one is in progress 

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Refactor download flow to eliminate extraneous selectors
* Force download rewind flow screen to wait to render until the initial request for progress has finished
* Lock button if another download is in progress

Screen Shot 2020-07-07 at 3 51 58 PM

#### Testing instructions

1. Open two different backups for the same site in side-by-side windows
2. Start downloadable file creation on one, reload the other window.
3. Verify the other button disables with "Another downloadable file is being created" until the first is complete
4. Create a fresh jetpack site ( can be JN ). Make sure the download page is able to load for a fresh backup purchase i.e. if there have been no previous downloads and the request for `/sites/:siteId/rewind/downloads` returns `[]`

Fixes 1164141197617539-as-1178122806736480

### CODE REVIEW

**Comment by mattgawarecki** [comment]
I was able to see the button change to "in progress" as described, so everything appeared to be working for me.

The issue @monsieur-z reported is definitely worth looking into, though. Are you able to see any errors/failures in your browser's Network panel?

**Comment by monsieur-z** [comment]
Neither in the console nor in the network panel.

File: client/state/data-layer/wpcom/sites/rewind/downloads/index.js [hunk 1]
```diff
@@ -81,8 +81,9 @@ const fromApi = ( data ) => ( {
  */
 export const updateProgress = ( { siteId }, apiData ) => {
 	const [ latestDownloadableBackup ] = apiData;
+
 	if ( isEmpty( latestDownloadableBackup ) ) {
-		return;
+		return updateRewindBackupProgress( siteId );
```
Comment (cbauerman) [inline client/state/data-layer/wpcom/sites/rewind/downloads/index.js:86]: an empty response is completely valid, indicates no proviso downloads

File: client/state/activity-log/backup/reducer.js [hunk 2]
```diff
@@ -43,16 +43,18 @@ export const backupProgress = keyedReducer( 'siteId', ( state = undefined, actio
...
-				backupPoint: action.backupPoint,
-				downloadId: action.downloadId,
-				progress: action.progress,
-				rewindId: action.rewindId,
-				startedAt: action.startedAt,
-				downloadCount: action.downloadCount,
-				validUntil: action.validUntil,
-				url: action.url,
-			};
+			return ! action.rewindId
```
Comment (cbauerman) [inline client/state/activity-log/backup/reducer.js:46]: updated to handle the new case below

**Comment by cbauerman** [comment]
> Neither in the console nor in the network panel.

This was because adding tracking to the request illuminated an issue with the handling of empty responses i.e. a site with no previous downloads. Should be fixed, but be aware I did rebase against master.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4075991

Thank you @cbauerman for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4075991).

**Pull request merged**